runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/businessmetrics/BusinessMetricsUtils.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.businessmetrics
+
+import aws.smithy.kotlin.runtime.InternalApi
+import aws.smithy.kotlin.runtime.collections.AttributeKey
+import aws.smithy.kotlin.runtime.collections.get
+import aws.smithy.kotlin.runtime.operation.ExecutionContext
+
+/**
+ * Keeps track of all business metrics along an operations execution
+ */
+@InternalApi
+public val businessMetrics: AttributeKey<MutableSet<String>> = AttributeKey("aws.sdk.kotlin#BusinessMetrics")
+
+/**
+ * If an endpoint is account ID based
+ */
+@InternalApi
+public val accountIdBasedEndPoint: AttributeKey<Boolean> = AttributeKey("aws.smithy.kotlin#AccountIdBasedEndpoint")
+
+/**
+ * If an endpoint is service endpoint override based
+ */
+@InternalApi
+public val serviceEndpointOverride: AttributeKey<Boolean> = AttributeKey("aws.smithy.kotlin#ServiceEndpointOverride")
```

Comment:
**Question**: I'm not sure I see the value of dedicated attribute keys for these. At the point we have access to the these attributes, we _could_ be calling `emitBusinessMetrics` already, right?

Reply:
Do you mean in the `DefaultEndpointProviderGenerator`? If so I don't think we can, we only have access to the endpoint attributes but not to the execution context attributes